### DISCUSSION TITLE
Return errors from property set / get

### DISCUSSION THREAD

**Discussion by Maddimax** [Kistler-Group/sdbus-cpp#238] [Q&A] [answered]
I would like to return errors from my get / set property functions. I'm using the IDL generated Adaptor classes, but I can't see any way to return an error state from those.

Am I missing something?

It would be great if I could for instance throw an sdbus::Error in the getter / setter, and have sdbus translate that into an error response.

**Comment by Maddimax** [comment, marked as answer]
Nevermind, I found that "SDBUS_THROW_ERROR" does the trick. Maybe documenting that makes sense?

**Reply by sangelovic** [reply]
The tutorial shows in several places (e.g. in a code snippet under https://github.com/Kistler-Group/sdbus-cpp/blob/master/docs/using-sdbus-c%2B%2B.md#providing-server-implementation-based-on-generated-adaptors) that server-side methods can simply throw `sdbus::Error` to signal/return errors to the caller.
